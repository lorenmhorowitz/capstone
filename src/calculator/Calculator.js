import Component from "react";

const square = 100;
class Calculator extends Component {
  // Shingles
  findBundles = sqFoot => {
    const bundlesPerSquare = 3 / square;
    const bundles = bundlesPerSquare * sqFoot;
    return bundles;
  };
  findShingles = sqFoot => {
    const shinglesPerBundle = 29;
    const shingles = shinglesPerBundle * sqFoot;
    return shingles;
  };
  findShingleNails = sqFoot => {
    const nailsPerSquare = 320 / square;
    const nails = nailsPerSquare * sqFoot;
    return nails;
  };
  // Ridges and Hips

  // Roll Roofing

  // Gutters and Eaves

  // Rake

  // Flashing

  // Step flashing
}

export default Calculator;
