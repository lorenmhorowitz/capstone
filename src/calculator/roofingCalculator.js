// Quantity Calculator
// Calculates the quantity of each building material

// Find number of 100 foot squares
const getNumberOfSquares = Measurements => {
  // Industry standard of 100 square feet per square
  return Math.ceil(
    getWasteFactorAdjustedMeasurements(Measurements).squareFootage / 100
  );
};

const getWasteFactorAdjustedMeasurements = Measurements => {
  const {
    squareFootage,
    ridgeLength,
    gutterLength,
    rakeLength,
    wasteFactor,
    stepFlashing
  } = Measurements;
  return {
    squareFootage: squareFootage + squareFootage * wasteFactor,
    ridgeLength: ridgeLength + ridgeLength * wasteFactor,
    gutterLength: gutterLength * (1 + wasteFactor),
    rakeLength: rakeLength * (1 + wasteFactor),
    wasteFactor: 0,
    length: stepFlashing + stepFlashing * wasteFactor
  };
};

// Bundles
const getBundleQuantity = Measurements => {
  // Industry standard of 3 bundles per square
  return getNumberOfSquares(Measurements) * 3;
};

// Shingles
const getShinglesQuantity = Measurements => {
  // Industry standard of 29 shingles per bundle
  return getBundleQuantity(Measurements) * 29;
};

const getNailsQuantity = Measurements => {
  return (
    getShinglesNailsQuantity(Measurements) +
    getUnderlaymentNailsQuantity(Measurements)
  );
};

// Nails for shingle
const getShinglesNailsQuantity = Measurements => {
  // Industry standard of 320 nails per square
  return 320 * getNumberOfSquares(Measurements);
};
// Ridges and Hips
const getCapShinglesQuantity = Measurements => {
  // Overlap of 5 inches per shingle, 12-5 = 7
  return Math.ceil(
    (12 *
      Math.ceil(getWasteFactorAdjustedMeasurements(Measurements).ridgeLength)) /
      7
  );
};

// Underlayment
const getUnderlaymentQuantity = Measurements => {
  // Industry standard of 3 feet by 33 feet, 3x33 = 99
  return Math.ceil(
    getWasteFactorAdjustedMeasurements(Measurements).squareFootage / 99
  );
};
// Underlayment Nails
const getUnderlaymentNailsQuantity = Measurements => {
  // Assuming 2 nails per square foot
  return (
    2 *
    Math.ceil(getWasteFactorAdjustedMeasurements(Measurements).squareFootage)
  );
};

// Gutters and Eaves
const getGuttersQuantity = Measurements => {
  // Returning the length of the Gutter and Eave
  // Sold per foot
  return getWasteFactorAdjustedMeasurements(Measurements).gutterLength;
};
// Rake
const getRakesQuantity = Measurements => {
  // Industry standard of 10.5 feet
  return Math.ceil(
    getWasteFactorAdjustedMeasurements(Measurements).rakeLength / 10.5
  );
};
// Flashing
const getFlashingQuantity = Measurements => {
  // Industry standard of 50 feet
  return Math.ceil(
    getWasteFactorAdjustedMeasurements(Measurements).ridgeLength / 50
  );
};
// Step flashing
const getStepFlashingQuantity = Measurements => {
  // Assume 7 inch with 2 inch overlap overlap, 12-7 = 5
  return Math.ceil(
    (getWasteFactorAdjustedMeasurements(Measurements).length * 12) / 5
  );
};

export default {
  getBundleQuantity,
  getCapShinglesQuantity,
  getFlashingQuantity,
  getGuttersQuantity,
  getNailsQuantity,
  getNumberOfSquares,
  getRakesQuantity,
  getShinglesQuantity,
  getStepFlashingQuantity,
  getUnderlaymentQuantity,
  getWasteFactorAdjustedMeasurements
};
