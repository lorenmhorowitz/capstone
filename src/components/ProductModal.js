import Modal from "@material-ui/core/Modal";
import ProductBox from "./ProductBox";
import ProductInfo from "./ProductInfo";
import PropTypes from "prop-types";
import React, { Component } from "react";
import { withStyles } from "@material-ui/core/styles";

const styles = {
  box: {
    height: "20%",
    width: "20%"
  },
  image: {
    height: "90%",
    marginLeft: "5px",
    marginTop: "20px",
    marginRight: "auto",
    width: "90%"
  },
  imageContainer: {
    float: "left",
    height: "40%",
    width: "40%"
  },
  modal: {
    background: "white",
    outline: "none",
    height: "75%",
    margin: "auto",
    marginTop: "5%",
    padding: "10px",
    width: "75%"
  },
  name: {
    fontSize: 20,
    marginTop: "3%",
    marginLeft: "10px",
    width: "50%"
  },
  price: {
    fontSize: 26,
    fontWeight: "bold"
  },
  productInfo: {
    float: "left",
    height: "100%",
    marginLeft: "30px",
    marginTop: "30px",
    width: "50%"
  },
  title: {
    fontSize: 22,
    fontWeight: "bold",
    marginTop: "2%",
    marginLeft: "18%"
  },
  otherProducts: {
    clear: "both",
    height: "100%",
    paddingLeft: "10px",
    paddingTop: "40px",
    width: "100%"
  }
};

class ProductModal extends Component {
  constructor(props) {
    super(props);

    this.state = {
      image: "",
      itemID: "",
      model: "",
      name: "",
      price: "",
      title: "",
      weight: ""
    };
  }

  componentWillMount() {
    this.setState({
      image: this.props.image,
      itemID: this.props.itemID,
      model: this.props.model,
      name: this.props.name,
      price: this.props.price,
      title: this.props.title,
      weight: this.props.weight
    });
  }

  changeModal = obj => {
    console.log("changing modal");
    console.log(obj);
    this.setState({
      image: obj.image,
      itemID: obj.itemID,
      model: obj.model,
      name: obj.name,
      price: obj.price,
      title: obj.brand,
      weight: obj.weight
    });
  };

  render() {
    const { classes } = this.props;

    let otherProducts = this.props.otherProducts;
    let index = 0;
    let productList = [];
    Object.keys(otherProducts).map(category => {
      Object.keys(otherProducts[category]).map(product => {
        if (
          this.props.itemID !==
          otherProducts[category][product].products[0].item_id
        ) {
          productList.push(
            <ProductBox
              key={index++}
              onChange={this.changeModal}
              product={otherProducts[category][product].products[0]}
              className={classes.box}
            />
          );
        }
      });
    });

    return (
      <Modal open={this.props.open} onClose={this.props.onClose}>
        <div className={classes.modal}>
          <div className={classes.container}>
            <div className={classes.imageContainer}>
              <img className={classes.image} src={this.state.image} alt="" />
            </div>
            <div className={classes.productInfo}>
              <ProductInfo
                itemID={this.state.itemID}
                model={this.state.model}
                name={this.state.name}
                price={this.state.price}
                title={this.state.title}
                weight={this.state.weight}
              />
            </div>
          </div>
          <div className={classes.otherProducts}>{productList.slice(0, 5)}</div>
        </div>
      </Modal>
    );
  }
}

ProductModal.propTypes = {
  classes: PropTypes.object.isRequired
};

export default withStyles(styles)(ProductModal);
