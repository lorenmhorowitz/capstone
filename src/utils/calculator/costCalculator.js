const getSubtotal = (cost, quantity) => {
  return cost * quantity;
};

export default {
  getSubtotal
};
