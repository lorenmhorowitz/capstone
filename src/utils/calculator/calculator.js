import roofingQuantityCalculator from "./roofingCalculator";
import costCalculator from "./costCalculator";

export default {
  ...roofingQuantityCalculator,
  ...costCalculator
};
