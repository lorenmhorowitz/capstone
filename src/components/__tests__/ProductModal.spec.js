import calculator from "../../utils/calculator/calculator";
import mockJobsDetailsList from "../../../mocks/mockJobsDetailsList";
import ProductModal from "../ProductModal";
import React from "react";
import { shallow } from "enzyme";

describe("Product modal", () => {
  it("Product modal test successful", () => {
    const currentProduct =
      mockJobsDetailsList.mockJobsDetailsList.results[0].products.roofing
        .shingles[0].products[0];
    const roofingQuantities = calculator.getRoofingProductQuantities(
      mockJobsDetailsList.mockJobsDetailsList.results[0].measurements
    );

    function testClose() {}

    let wrapper = shallow(
      <ProductModal
        brand={currentProduct.brand}
        image={currentProduct.image}
        itemID={currentProduct.itemID}
        model={currentProduct.model}
        name={currentProduct.name}
        open={"false"}
        onClose={testClose}
        otherProducts={currentProduct.otherProducts}
        price={currentProduct.price}
        quantity={currentProduct.quantity}
        roofingQuantity={roofingQuantities}
        weight={currentProduct.weight}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
