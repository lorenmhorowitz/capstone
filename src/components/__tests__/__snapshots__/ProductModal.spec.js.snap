// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product modal Product modal test successful 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WithStyles(ProductModal)
    brand="Owens Corning"
    image="https://images.homedepot-static.com/productImages/271746c1-8a65-4cd8-86d7-bdd4e80e9f5e/svn/onyx-black-owens-corning-roof-shingles-786353-64_1000.jpg"
    model="786353"
    name="Oakridge Algae Resistant Onyx Black Laminate Architectural Shingles (32.8 sq. ft. per Bundle)"
    onClose={[Function]}
    open="false"
    price={27}
    quantity={3}
    roofingQuantity={
      Object {
        "dripEdge": 150,
        "nails": 6,
        "ridgeCaps": 119,
        "shingles": 108,
        "underlayment": 37,
      }
    }
    weight={13.5}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "brand": "Owens Corning",
      "classes": Object {
        "box": "ProductModal-box-1",
        "brand": "ProductModal-brand-2",
        "image": "ProductModal-image-3",
        "imageContainer": "ProductModal-imageContainer-4",
        "modal": "ProductModal-modal-5",
        "name": "ProductModal-name-6",
        "otherProducts": "ProductModal-otherProducts-9",
        "price": "ProductModal-price-7",
        "productInfo": "ProductModal-productInfo-8",
      },
      "image": "https://images.homedepot-static.com/productImages/271746c1-8a65-4cd8-86d7-bdd4e80e9f5e/svn/onyx-black-owens-corning-roof-shingles-786353-64_1000.jpg",
      "itemID": undefined,
      "model": "786353",
      "name": "Oakridge Algae Resistant Onyx Black Laminate Architectural Shingles (32.8 sq. ft. per Bundle)",
      "onClose": [Function],
      "open": "false",
      "otherProducts": undefined,
      "price": 27,
      "quantity": 3,
      "roofingQuantity": Object {
        "dripEdge": 150,
        "nails": 6,
        "ridgeCaps": 119,
        "shingles": 108,
        "underlayment": 37,
      },
      "weight": 13.5,
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "brand": "Owens Corning",
        "classes": Object {
          "box": "ProductModal-box-1",
          "brand": "ProductModal-brand-2",
          "image": "ProductModal-image-3",
          "imageContainer": "ProductModal-imageContainer-4",
          "modal": "ProductModal-modal-5",
          "name": "ProductModal-name-6",
          "otherProducts": "ProductModal-otherProducts-9",
          "price": "ProductModal-price-7",
          "productInfo": "ProductModal-productInfo-8",
        },
        "image": "https://images.homedepot-static.com/productImages/271746c1-8a65-4cd8-86d7-bdd4e80e9f5e/svn/onyx-black-owens-corning-roof-shingles-786353-64_1000.jpg",
        "itemID": undefined,
        "model": "786353",
        "name": "Oakridge Algae Resistant Onyx Black Laminate Architectural Shingles (32.8 sq. ft. per Bundle)",
        "onClose": [Function],
        "open": "false",
        "otherProducts": undefined,
        "price": 27,
        "quantity": 3,
        "roofingQuantity": Object {
          "dripEdge": 150,
          "nails": 6,
          "ridgeCaps": 119,
          "shingles": 108,
          "underlayment": 37,
        },
        "weight": 13.5,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
